### ISSUE TITLE
Java 8 Dates support

### ISSUE THREAD

**Issue by danielraczynski** [benas/random-beans#41]
Since I'm getting the following error 

Caused by: java.lang.NoSuchMethodException: java.time.ZonedDateTime.<init>()
	at java.lang.Class.getConstructor0(Class.java:3082)
	at java.lang.Class.newInstance(Class.java:412)
	... 10 more

I assume that Java 8 Dates types are not supported (the same applies for LocalDateTime and OffsetDateTime).
I also saw that for deprecated Date type the library works fine.

Didn't have time to look at the code - but if this is the case than I advice you to add it to future release since this is the standard for Java 8.

**Comment by benas** [comment]
Java 8 Date types are not supported yet.
I'll add support for them in a future release.

Thank you for pointing out this issue.

Regards
Mahmoud

**Comment by benas** [comment]
Hi @danielraczynski ,

I added support for Java 8 Date and Time types in the 2.0.0-SNAPSHOT version (which I deployed to maven central repo). Use the following repo to get it:

```xml
<repository>
    <id>ossrh</id>
    <url>https://oss.sonatype.org/content/repositories/snapshots</url>
</repository>
```

It would be great if you could give it a try and tell me if it is ok for you.
Thank you.

Kind regards
Mahmoud

**Comment by prietopa** [comment]
Hi, i dont unferstand because in #28,  you said that be carefull to use
Java8 classes as Supplier class.

JMPrieto.

**Comment by benas** [comment]
Hello @prietopa 

Only module `random-beans-java8` requires Java 8, all other modules are Java 6 compatible.
In #28, the goal is to add a method in the `PopulatorBuilder` that accepts a `Supplier` an adapt it to a `Randomizer` (which is a great idea), but this requires the `random-beans` module to use Java 8 which makes the library incompatible with Java 6..

The goal is to use Java 8 APIs only in the `random-beans-java8` and keep using Java 6 compatible APIs in other modules.

Java is backward compatible, unless you use APIs introduced in a newer version..
I've tried to set compiler settings (-source 1.8, -target 1.6) to compile in version 8 and produce version 6 byte code, but this is not possible.

**Comment by prietopa** [comment]
understood!

JMPrieto

2015-12-22 11:01 GMT+01:00 Mahmoud Ben Hassine <notifications@github.com>:

**Comment by wheredevel** [comment]
@benas I cloned the repo and noticed java8 module commented out.
Uncommented, and it fails to build.
 What's missing? Perhaps I could help...

**Comment by benas** [comment]
I decided to postpone Java 8 support to v3. See here: https://github.com/benas/random-beans/issues/49#issuecomment-174336657

This is because I was not able to make the code base compile with -source=1.8 and -target=1.7

**Comment by benas** [comment]
Hi @danielraczynski 

As discussed, Java 8 support has been postponed to v3 (the upcoming release). I have just published the 3.0.0-SNAPSHOT to maven central. This version includes support for JSR 310 types.

Here is a [sample bean](https://github.com/benas/random-beans/blob/master/random-beans-java8/src/test/java/io/github/benas/randombeans/randomizers/java8/beans/ThreeTenBean.java#L29) and a [passing test](https://github.com/benas/random-beans/blob/master/random-beans-java8/src/test/java/io/github/benas/randombeans/randomizers/java8/ThreeTenSupportTest.java#L45). It would be great if you could give it a try and tell me if it is ok for you. Thank you.

Kind regards
Mahmoud

**Comment by benas** [comment]
hello @danielraczynski 

If you have any problem with JSR 310 types support, please don't hesitate to file an issue.

Kind regards
Mahmoud
